docs/content/docs/patterns/observability/tracing.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "Tracing"
+description: "Tracing"
+type: docs
+---
+
+# Tracing
+Open Service Mesh (OSM) allows optional deployment of Jaeger for tracing. 
+
+## Jaeger
+[Jaeger](https://www.jaegertracing.io/) is an open source tracing system used for monitoring and troubleshooting distributed systems. It can be deployed with OSM as a new instance or you may bring your own instance. 
+
+### Automatic bring up
+By default, Jaeger deployment and tracing as a whole is disabled.
+
+Jaeger can be automatically deployed by using the `--deploy-jaeger` OSM CLI flag at install time or by toggling the `deployJaeger` value in `values.yaml`.
+
+Similarly, tracing can be enabled and customized during installation (`tracing` section in `values.yaml`) or at runtime by editing the `osm-config` ConfigMap. Tracing can be enabled, disabled and configured at any time to support BYO scenarios.
+
+### BYO (Bring-your-own)
+The following section documents the additional steps needed to allow an already running instance of Jaeger to integrate with your OSM control plane.
+
+#### Prerequisites
+* A running Jaeger instance
+
+The sections below outline how to make required updates depending on whether you already have a running instance of OSM or are installing OSM for the first time. In either case, the following `tracing` values in `values.yaml` are being updated to point to your Jaeger instance:
+1. `enable`: set to `true` to tell the Envoy connection manager to send tracing data to a specific address (cluster)
+1. `address`: set to the destination cluster of your Jaeger instance
+1. `port`: set to the destination port for the listener that you intend to use
+1. `endpoint`: set to the destination's API or collector endpoint where the spans will be sent to
+
+
+#### Deploying Jaeger to a running instance of the OSM control plane
+
+If you already have OSM running, `tracing` values must be updated in the OSM ConfigMap using:
+
+```console
+osm mesh upgrade --enable-tracing --tracing-address <tracing server hostname> --tracing-port <tracing server port> --tracing-endpoint <tracing server endpoint>
```

Comment:
I couldn't find too much docs on `osm mesh upgrade` since it's pretty new. It would be good to add:

> NOTE : This command upgrades an OSM control plane configuration by upgrading the
underlying Helm release.

Can this override user configured changes in the OSM configmap?

Reply:
What if users don't want to upgrade their helm release? They could technically just `kubectl edit`.

Reply:
> Can this override user configured changes in the OSM configmap?

Yes, any values supplied to the `osm mesh upgrade` command will override the current state. When a value is not specified, it is left unchanged _from the previous Helm release_. So If a user does `osm install --enable-tracing`, then `kubectl edit` to disable it, then `osm mesh upgrade`, tracing will be enabled. If `osm mesh upgrade --enable-tracing=false` was used instead of `kubectl edit`, the subsequent `osm mesh upgrade` would keep it disabled.

> What if users don't want to upgrade their helm release? They could technically just `kubectl edit`.

Yes, that's correct. Users shouldn't expect those changes to persist after an upgrade though.

Reply:
I've added the note. Is there anything I need to update about using `kubectl edit` or values persisting?

Reply:
For values persisting, if that's not clear between the upgrade guide and the CLI `--help` text then we should edit those.

Reply:
> For values persisting, if that's not clear between the upgrade guide and the CLI osm mesh upgrade --help text then we should edit those.

Yes that would be great. I just looked at `osm mesh upgrade -h` and `osm install -h`, there is nothing about value override for user-provided values. Would be good to add it. 

There is this for persisting flags provided by helm values:
> By default, if flags tied to Helm chart values are not specified for "osm mesh
upgrade", then their current value will be carried over to the new release.

So if these tracing flags are provided as part of `osm mesh upgrade` then it is safe to assume they will be persisted/carried over in future releases.

Reply:
> Yes that would be great. I just looked at `osm mesh upgrade -h` and `osm install -h`, there is nothing about value override for user-provided values. Would be good to add it.

@ritazh Just opened #2518, PTAL